xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/migration/AbstractDataMigrationManager.java

```diff
@@ -347,12 +347,18 @@ public void initialize() throws InitializationException
             } else {
                 Set<String> ignoredMigrations = new HashSet<>(this.hibernateConfiguration.getIgnoredMigrations());
                 for (DataMigration migrator : getAllMigrations()) {
+                    XWikiDBVersion migratorVersion = migrator.getVersion();
                     if (ignoredMigrations.contains(migrator.getClass().getName())
-                        || ignoredMigrations.contains(migrator.getVersion().toString())) {
+                        || ignoredMigrations.contains(migratorVersion.toString())) {
                         continue;
                     }
                     XWikiMigration migration = new XWikiMigration(migrator, false);
-                    availableMigrations.put(migrator.getVersion(), migration);
+                    if (!availableMigrations.containsKey(migratorVersion)) {
+                        availableMigrations.put(migratorVersion, migration);
+                    } else {
+                        throw new InitializationException(
+                            String.format("A migration with version [%s] already exists.", migratorVersion));
```

Comment:
Giving a bit more information, like the classes of the two colliding migrations, would be nice.

Reply:
Ok will improve the message. Besides that this check is enough you think? Was wondering if checking versions of forced migration was needed too?

Reply:
So I actually changed a bit the logic to perform a check of all migrations directly at the beginning, by getting list of components. It involves one more loop over the list of components that I didn't have before, but it allows to check globally the migrations.

Reply:
Also I'm just realizing that previous implem was wrong: in case of forced migration you'd have had an exception thrown actually...